tensorboard/plugins/debugger_v2/debug_data_multiplexer.py

```diff
@@ -446,6 +451,104 @@ def GraphExecutionData(self, run, begin, end, trace_id=None):
             ],
         }
 
+    def GraphOpInfo(self, run, graph_id, op_name):
+        """Get the information regarding a graph op's creation.
+
+        Args:
+          run: Name of the run.
+          graph_id: Debugger-generated ID of the graph that contains
+            the op in question. This ID is available from other methods
+            of this class, e.g., the return value of `GraphExecutionDigests()`.
+          op_name: Name of the op.
+
+        Returns:
+          A JSON-serializable object containing the information regarding
+          the op's creation.
+
+        Raises:
+          NotFoundError if the graph_id or op_name does not exist.
+        """
+        runs = self.Runs()
+        if run not in runs:
+            return None
+        try:
+            graph = self._reader.graph_by_id(graph_id)
+        except KeyError:
+            raise errors.NotFoundError(
+                'There is no graph with ID "%s"' % graph_id
+            )
+        try:
+            # TODO(cais): Use public get_op_creation_digest() when available.
+            op_creation_digest = graph._op_by_name[op_name]
+        except KeyError:
+            raise errors.NotFoundError(
+                'There is no op named "%s" in graph with ID "%s"'
+                % (op_name, graph_id)
+            )
+        data_object = self._opCreationDigestToDataObject(op_creation_digest)
+        # Populate data about immediate inputs.
+        data_object["inputs"] = []
+        if op_creation_digest.input_names:
+            data_object["inputs"] = [None] * len(op_creation_digest.input_names)
+            for slot, input_tensor_name in enumerate(
+                op_creation_digest.input_names
+            ):
+                input_op_name = tensor_name_to_op_name(input_tensor_name)
+                input_op_digest = graph._op_by_name.get(input_op_name, None)
+                # TODO(b/155308456): Once Placeholder and Const ops have their
+                # ad hoc naming issues fixed, remove this `if` logic.
+                if input_op_digest:
+                    data_object["inputs"][
+                        slot
+                    ] = self._opCreationDigestToDataObject(input_op_digest)
+        # Populate data about immediate consuming ops.
+        data_object["consumers"] = []
+        op_creation_digests = list(graph._op_by_name.values())
+        for digest in op_creation_digests:
+            if digest.input_names and any(
+                tensor_name_to_op_name(input_name) == op_creation_digest.op_name
+                for input_name in digest.input_names
+            ):
+                data_object["consumers"].append(
+                    self._opCreationDigestToDataObject(digest)
+                )
+        return data_object
+
+    def _opCreationDigestToDataObject(self, op_creation_digest):
```

Comment:
This function name sounds like it's just a format conversion, when it's actually reading from disk.  Maybe `_loadOpDataObject` or similar?